tests/e2e/support/objects/app-files/share/actions.ts

```diff
@@ -60,21 +66,70 @@ export const createShare = async (args: createShareArgs): Promise<void> => {
 
 /**/
 
-export type ShareStatusArgs = Omit<ShareArgs, 'recipients'>
+export interface ShareStatusArgs extends Omit<ShareArgs, 'recipients'> {
+  via?: 'STATUS' | 'CONTEXT_MENU'
+}
 
 export const acceptShare = async (args: ShareStatusArgs): Promise<void> => {
-  const { resource, page } = args
-  await Promise.all([
-    page.locator(util.format(shareAcceptDeclineButton, resource, 'status-accept')).click(),
-    page.waitForResponse((resp) => resp.url().includes('shares') && resp.status() === 200),
-    page.locator(util.format(filesSharedWithMeAccepted, resource)).waitFor()
-  ])
+  const { resource, via, page } = args
+  if (via === 'CONTEXT_MENU') {
+    await clickActionInContextMenu({ page, resource }, 'accept-share')
+  } else {
+    await page.locator(util.format(shareAcceptDeclineButton, resource, 'status-accept')).click()
+    await page.waitForResponse(
+      (resp) =>
+        resp.url().includes('shares') && resp.status() === 200 && resp.request().method() == 'POST'
+    )
+  }
+  await page.locator(util.format(filesSharedWithMeAccepted, resource)).waitFor()
 }
 
 export const declineShare = async (args: ShareStatusArgs): Promise<void> => {
+  const { page, resource, via } = args
+  if (via === 'CONTEXT_MENU') {
```

Comment:
```suggestion
  if (via === MENU.CONTEXT_MENU) {
```
Seems like this is used in multiple places. Would be better if we take the value from some variables.